earth2mip/forecast_metrics_io.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Routines for reading and writing forecast metrics to a directory of csv files.
+
+The csv files contain the records::
+
+    initial_time_iso, lead_time_hours, channel, metric, value
+    2022-01-01T00:00:00,24,t2m,rmse,25.6
+
+"""
+import csv
+from typing import IO
+import datetime
+
+import os
+import pandas as pd
+
+
+def read_metrics(directory: str) -> pd.Series:
```

Comment:
This introduces another datatype into the over all dataflow of scoring.Curious about reasoning for pd Series over xarray since sometimes this just gets converted to xarray dataset.

Reply:
this is as good point. While I would like everyone who uses xarray to also know pandas, this is probably not true.

however this is meant to be low-level function to help open the data and not much more, and pandas felt like a tighter fit. The issue with xarray is that series.to_xarray() will actually expand all the indices (init_time, lead_time, channel, metric) into an array, filling in any NaNs if there aren't any records. Imagine some metrics only happen on some channels.

Overall, recent work has minimized xarray in the code (i.e. data sources return numpy arrays).

Reply:
Makes sense, not really a problem if its low level and not a API we expect users to interface with. :+1: